ts/components/messages/MessageDueDateBar.tsx

```diff
@@ -57,182 +70,210 @@ const styles = StyleSheet.create({
+    : status === "valid"
+    ? customVariables.colorWhite
+    : assertNever(status);
+
+// The calendar icon is shown if:
+// - the payment related to the message is not yet paid
+// - the message has a due date
+const CalendarIcon: React.FunctionComponent<{
+  status: PaymentStatus;
+  dueDate: Date;
+}> = ({ status, dueDate }) => {
+  const iconBackgoundColor = getCalendarIconBackgoundColor(status);
 
-  // The calendar icon is shown if:
-  // - the payment related to the message is not yet paid
-  // - the message has a due date
-  private renderCalendarIcon = () => {
-    const { dueDate: maybeDueDate } = this;
-    return maybeDueDate.fold(null, dd => {
-      const iconBackgoundColor = this.paid
-        ? customVariables.lighterGray
-        : this.isPaymentExpiring || this.isPaymentExpired
-        ? customVariables.colorWhite
-        : customVariables.brandDarkGray;
-
-      const textColor = this.paid
-        ? customVariables.colorWhite
-        : this.isPaymentExpiring || !this.isPaymentExpirable
-        ? customVariables.calendarExpirableColor
-        : this.isPaymentExpired && this.isPaymentExpirable
-        ? customVariables.brandDarkGray
-        : customVariables.colorWhite;
-
-      return (
-        <CalendarIconComponent
-          month={capitalize(formatDateAsMonth(dd))}
-          day={formatDateAsDay(dd)}
-          backgroundColor={iconBackgoundColor}
-          textColor={textColor}
-        />
-      );
-    });
-  };
+  const textColor = getCalendarTextColor(status);
 
+  return (
+    <CalendarIconComponent
+      month={capitalize(formatDateAsMonth(dueDate))}
+      day={formatDateAsDay(dueDate)}
+      backgroundColor={iconBackgoundColor}
+      textColor={textColor}
+    />
+  );
+};
+
+const bannerStyle = (status: PaymentStatus): ViewStyle =>
+  status === "paid"
+    ? { backgroundColor: customVariables.brandGray }
+    : status === "expiredAndExpirable"
+    ? { backgroundColor: customVariables.brandDarkGray }
+    : status === "expiredNotExpirable"
+    ? { backgroundColor: customVariables.calendarExpirableColor }
+    : status === "expiring"
+    ? { backgroundColor: customVariables.calendarExpirableColor }
+    : status === "valid"
+    ? { backgroundColor: customVariables.brandGray }
+    : assertNever(status);
+
+const isPaymentExpirable = (message: CreatedMessageWithContent): boolean =>
+  paymentExpirationInfo(message).fold(false, isExpirable);
+
+const isPaymentExpired = (message: CreatedMessageWithContent): boolean =>
+  paymentExpirationInfo(message).fold(false, isExpired);
+
+const isPaymentExpiring = (message: CreatedMessageWithContent): boolean =>
+  paymentExpirationInfo(message).fold(false, isExpiring);
+
+const paid = (payment: PaidReason | undefined): boolean =>
+  payment !== undefined;
+
+const calculatePaymentStatus = (
+  payment: PaidReason | undefined,
+  message: CreatedMessageWithContent
+): PaymentStatus =>
+  isPaymentExpired(message) && isPaymentExpirable(message)
```

Comment:
Use if/else in this case is more readable.
If we add another state this code will increase